Check for computables when generating update types

dardanbujupaj opened the pull request:
This PR adds a check for computed fields before generating update types.

This should fix https://github.com/edgedb/edgedb/issues/4230 as empty update types can't be generated anymore. 
Unfortunately I'm not able to test it properly as I have issues getting the local development server working...

dardanbujupaj commented:
I just realized, that a similar check should probably be introduced for the Insert types. But I'm a bit confused, as there's not even a check regarding the readonly properties.

vpetrovykh commented:
We definitely want a test in the test suite that actually reproduces the #4230 bug so that we can be sure that this fixes it as well as that future changed don't break it again.

Whether we need a similar check for insert types depends on whether this bug is reproducible in Insert types.

So @dardanbujupaj, if you feel up to it, you can come back after you figure out what went wrong with your local development server and write a test for this.

dardanbujupaj commented:
@vpetrovykh Finally got the test environment to work.

I updated the PR, but instead of adding explicit tests, I updated `tests/schemas/graphql_schema.esdl` to contain a type according to the description in https://github.com/edgedb/edgedb/issues/4230.

This leads to all `graphql_schema`-tests failing and the changes to `edb/graphql/types.py` resolve this as expected.

You can check the failing tests at commit https://github.com/edgedb/edgedb/pull/4231/commits/5db40e0fa6a1e0a0d0ab996f73676ef7bb6c9d5a with  `edb test -k test_graphql_schema`.

dardanbujupaj commented:
This seems to be fixed in recent releases 👍🏻
